src/modules/triple-store/implementation/ot-triple-store.js

```diff
@@ -346,32 +343,45 @@ class OtTripleStore {
         await this.queryVoid(repository, query);
     }
 
-    async getKnowledgeAssetNamedGraph(repository, ual) {
-        const query = `
-            PREFIX schema: <${SCHEMA_CONTEXT}>
-            CONSTRUCT  { ?s ?p ?o } 
-            WHERE {
-                GRAPH <${ual}> {
-                    ?s ?p ?o .
-                }
-            }
-        `;
+    async getKnowledgeAssetNamedGraph(repository, ual, visibility) {
+        let whereClause;
 
-        return this.construct(repository, query);
-    }
+        switch (visibility) {
+            case TRIPLES_VISIBILITY.PUBLIC:
+            case TRIPLES_VISIBILITY.PRIVATE:
+                whereClause = `
+                    WHERE {
+                        GRAPH <${ual}/${visibility}> {
+                            ?s ?p ?o .
+                        }
+                    }
+                `;
+                break;
+            case TRIPLES_VISIBILITY.ALL:
+                whereClause = `
```

Comment:
Is this better than just having filter by ual?

Reply:
You are correct I'll rewrite  the query

Reply:
My only concern is that it's maybe faster to just provide names than parse the name like strings